.azure-pipelines.yml

```diff
@@ -57,7 +57,7 @@ jobs:
           set -e
           uv pip install --system --compile tomli packaging
           deps=`python3 ci/scripts/min-deps.py pyproject.toml --extra dev test`
-          uv pip install --system --compile $deps pytest-cov "anndata @ ."
+          uv pip install --system --compile $deps pytest-cov "anndata @ .[test,dev]"
```

Comment:
@flying-sheep I think this is a bug, no? The min-deps are created without respect for upper bounds so the constraints can end up not including the upper bounds because the `min-deps` script doesn't care about it? I think this is the correct fix given the fact that the script is called "min-deps" and not "min-deps-with-upper-bound" but maybe I'm off